src/components/Swap/SwapActionButton/index.tsx

```diff
@@ -36,6 +36,8 @@ export default function SwapActionButton() {
 
   if (!account || !isActive) {
     return <ConnectWalletButton />
+  } else if (error === ChainError.UNSUPPORTED_CHAIN) {
+    return <SwitchChainButton chainId={SupportedChainId.MAINNET} />
```

Comment:
I think this is an improvement, but I think the expected behavior is that the target chain should match the tokens passed into the widget. I.e., if the user is connected to (unsupported network) and the token input/outputs are on Polygon, then the widget should say "switch to polygon"